glean-core/android/src/main/java/mozilla/telemetry/glean/Glean.kt

```diff
@@ -166,7 +166,13 @@ open class GleanInternalAPI internal constructor () {
 
             // If any pings were registered before initializing, do so now.
             // We're not clearing this queue in case Glean is reset by tests.
-            pingTypeQueue.forEach { registerPingType(it) }
+            synchronized(this@GleanInternalAPI) {
+                pingTypeQueue.forEach {
+                    LibGleanFFI.INSTANCE.glean_register_ping_type(
```

Comment:
Why can't we call `registerPingType` directly here? We should already hold the mutex at this point so it should be fine.

I'm fine with keeping the code as it is btw, But if there's a specific reason why we can't do that, please let's add a comment about it here

Reply:
Good catch -- I was playing with that to avoid the issue and it ultimately didn't matter.  I agree "using the real API" is better.  Changing it back...